libcnb-cargo/src/main.rs

```diff
@@ -223,3 +379,332 @@ fn calculate_dir_size(path: impl AsRef<Path>) -> io::Result<u64> {
 
     Ok(size_in_bytes)
 }
+
+fn get_buildpack_workspace() -> BuildpackWorkspace {
+    let workspace_root = get_workspace_root();
+    let workspace_cargo_metadata = get_cargo_metadata(&workspace_root);
+    let workspace_target_directory = workspace_cargo_metadata
+        .target_directory
+        .into_std_path_buf();
+    BuildpackWorkspace {
+        root: workspace_root,
+        target_dir: workspace_target_directory,
+    }
+}
+
+#[allow(clippy::too_many_lines)]
+fn get_buildpack_projects(
+    buildpack_workspace: &BuildpackWorkspace,
+    args: &PackageArgs,
+) -> Vec<BuildpackProject> {
+    let mut buildpack_projects: Vec<BuildpackProject> = vec![];
+
+    for buildpack_dir in find_buildpack_directories(buildpack_workspace) {
```

Comment:
I think this function would greatly benefit from being refactored:

- Introduce a function for "getting" one buildpack project by path, this will enable us to replace the `continue`s with `?` ([`Try`](https://doc.rust-lang.org/std/ops/trait.Try.html)) which read top-to-bottom.
- Introduce an error enum for "getting" a buildpack project. This way, the errors can be handled (even if that "handling" is just filtering failed ones and logging) later on, decluttering the code.
- I mentioned this in the `Buildpackage` definition already, changing it to contain parsed Uris makes this code a little bit simpler.